### ISSUE TITLE
Enable managed prometheus disables cloud monitoring in v28

### ISSUE THREAD

**Issue by szihai** [terraform-google-modules/terraform-google-kubernetes-engine#1786]
### TL;DR

Shouldn't that be left as default value "System"?

### Expected behavior

Shouldn't that be left as default value "System"?

### Observed behavior

In v27, when managed prometheus is enabled, the cloud monitoring value was left as default:
![image](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/assets/26101651/6e78a84b-974d-4621-a192-80c1e8c622d6)
Now in v28, the cloud monitoring is disabled:
![image](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/assets/26101651/e8002401-b9a8-4033-9082-9a761ff4508f)
Our configuration did not change. Also when checking with GCP documentation, the default value should still be "System". 
![image](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/assets/26101651/2855b260-4416-47c3-a381-98e3b076fe1d)

Please fix this issue.

### Terraform Configuration

```hcl
module "gke" {
  source  = "terraform-google-modules/kubernetes-engine/google//modules/beta-private-cluster"
  version = "~> 28.0.0"

  ...

  #managed prometheus
  monitoring_enable_managed_prometheus = var.monitoring_enable_managed_prometheus
  ...
}

variable "monitoring_enable_managed_prometheus" {
  type        = bool
  description = "Configuration for Managed Service for Prometheus. Whether or not the managed collection is enabled."
  default     = true
}
```

### Terraform Version

```sh
Terraform v1.6.1
on linux_amd64
```

**Comment by szihai** [comment]
I stay corrected. The version that works is NOT v27.0.0, but the main branch two weeks ago. So does it mean it will be fixed in the next release?

**Comment by szihai** [comment]
I tested on the commit 2beb7207. Disregard the setting of "monitoring_enable_managed_prometheus", Terraform plan only wants to disable cloud monitoring:
![image](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/assets/26101651/e6dd75fd-c344-4aa5-be0a-2a457ad7cb92)
Which means the release candidate for v29.0.0 has not fixed this issue.

**Comment by szihai** [comment]
I just tried with v29.0.0, the problem is still there. It disables cloud monitoring when enables prometheus.

**Comment by m3adow** [comment]
This issue is still a problem (together with most other GMP issues).

**Comment by bergemalm** [comment]
Still an issue!

**Comment by m0ps** [comment]
It's not really an issue. The fix is pretty simple:
define `monitoring_enabled_components` variable.

Source: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/1706#issuecomment-1810686466

**Comment by m3adow** [comment]
> It's not really an issue. The fix is pretty simple: define `monitoring_enabled_components` variable.
> 
> Source: [#1706 (comment)](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/1706#issuecomment-1810686466)

Sadly that's not an option for some cluster variants, as they don't (yet?) support `monitoring_enabled_components`: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/issues/1793
